Add some more information to the readme (#8)
chaiNNer-org/spandrel: README.md

# Spandrel

[![PyPI package](https://img.shields.io/badge/pip%20install-spandrel-brightgreen)](https://pypi.org/project/spandrel/)
[![version number](https://img.shields.io/pypi/v/spandrel?color=green&label=version)](https://github.com/chaiNNer-org/spandrel/releases)
![PyPi Downloads](https://img.shields.io/pypi/dw/spandrel)
![PyPi Wheel](https://img.shields.io/pypi/wheel/spandrel)

> ![Python Version](https://img.shields.io/pypi/pyversions/spandrel)

Reviewer: I don't like that these 3 don't link to anything. When you click on them, you'll just see the image.

I'm also not sure about "wheel | yes". What does that even mean?

Author: Some packages release things other than wheels. I agree it isn't that useful, I only did it cuz syrupy did it

Reviewer: > Some packages release things other than wheels.

This doesn't really apply to us, so I think we can remove it.

Author: Neither of these are really meant to link to anything, they're just visual information. But i'll add links to stuff anyway

- [FBCNN](https://github.com/jiaxi-jiang/FBCNN) | [Models](https://github.com/jiaxi-jiang/FBCNN/releases/tag/v1.0)

## File type support

> Spandrel mainly supports loading .pth files for its supported architectures. This is what you will typically find from official repos and community trained models. However, Spandrel also supports loading TorchScript traced models (.pt) by extracting the weights out of them, as well as loading certain types of .ckpt files, as well as any supported model that has been saved as or converted to a .safetensors file.

Reviewer: 1. as well, as well
2. I don't think we need to mention *how* we support these file formats.

```suggestion
Spandrel mainly supports loading `.pth` files for all supported architectures. This is what you will typically find from official repos and community trained models. However, Spandrel also supports loading TorchScript traced models (`.pt`), certain types of `.ckpt` files, as well as any supported model that has been saved as or converted to a `.safetensors` file.
```

## Security

> As you may or may not know, loading .pth files usually poses a security risk due to python's pickle module being unsafe and vulnerable to arbitrary code execution. Because of this, Spandrel uses a custom unpickler function that only allows loading certain types of data out of a .pth file, meaning that ACE is not an issue.

Reviewer: 1. Let's assume only the best of the user and say that they know about it.
2. I want to directly say that we prevent ACE.

```suggestion
As you may know, loading `.pth` files usually [poses a security risk](https://github.com/pytorch/pytorch/issues/52596) due to python's `pickle` module being unsafe and vulnerable to arbitrary code execution (ACE). Because of this, Spandrel uses a custom unpickler function that only allows loading certain types of data out of a .pth file. This completely prevents ACE and makes loading untrusted files secure.
```
